src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,940 @@
+      fm = gen_mat_iso(G.deg, G.ring)
+      G.ring_iso=fm.fr
+      G.mat_iso=fm
+
+   elseif sym === :X
+      if isdefined(G,:X) return getfield(G,:X) end
+      if isdefined(G,:descr)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         assign_from_description(G)
+      elseif isdefined(G,:gens)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         V = GAP.julia_to_gap([g.X for g in G.gens])
+         G.X=GAP.Globals.Group(V)
+      end
+
+   elseif sym === :gens
+      if isdefined(G, :gens) return getfield(G, :gens) end
+      if !isdefined(G, :X)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         assign_from_description(G)
+      end
+      L = GAP.Globals.GeneratorsOfGroup(getfield(G,:X))
+      G.gens=[MatrixGroupElem(G,G.mat_iso(L[i]),L[i]) for i in 1:length(L)]
+
+   else
+      return getfield(G, sym)
+
+   end
+
+end
+
+
+# add field :X
+function Base.getproperty(x::MatrixGroupElem, sym::Symbol)
+   if sym === :X && !isdefined(x,:X)
+      if !isdefined(x.parent,:ring_iso)
+         fm = gen_mat_iso(x.parent.deg, x.parent.ring)
+         setfield!(x.parent,:ring_iso,fm.fr)
+         setfield!(x.parent,:mat_iso,fm)
+      end
+      setfield!(x, :X, x.parent.mat_iso(x.elm))
+   elseif sym == :elm && !isdefined(x, :elm)                        # assuming that x.X, hence x.parent.ring_iso, are defined
+      setfield!(x, :elm, x.parent.mat_iso(x.X))
+   end
+   return getfield(x,sym)
+end
+
+Base.IteratorSize(::Type{<:MatrixGroup}) = Base.SizeUnknown()
+
+function Base.iterate(G::MatrixGroup)
+  L=GAP.Globals.Iterator(G.X)
+  if GAP.Globals.IsDoneIterator(L)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(L)
+  return MatrixGroupElem(G, G.mat_iso(i),i), L
+end
+
+function Base.iterate(G::MatrixGroup, state)
+  if GAP.Globals.IsDoneIterator(state)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(state)
+  return MatrixGroupElem(G, G.mat_iso(i),i), state
+end
+
+########################################################################
+#
+# Membership
+#
+########################################################################
+
+
+function ==(G::MatrixGroup,H::MatrixGroup)
+   G.deg==H.deg || return false
+   G.ring==H.ring || return false
+   if isdefined(G, :descr) && isdefined(H, :descr)
+      return G.descr == H.descr
+   end
+   if isdefined(G, :gens) && isdefined(H, :gens)
+      if G.gens==H.gens return true  end       #TODO this works only if the == between elements does not keep track of the parent
+   end
+   return G.X==H.X
+end
+
+
+# this saves the value of x.X
+# x_gap = x.X if this is already known, x_gap = nothing otherwise
+function lies_in(x::MatElem, G::MatrixGroup, x_gap)
```

Comment:
1. Yes, that's a good idea, and it could be done for isometry groups too. Though, I would expect GAP already uses the methods of the determinant to evaluate membership in GL and SL; the contrary would be very strange.
2. That's why I was considering the idea that if `true` is returned by the line `G==H`, then the defined fields of `G` should be assigned to `H` and viceversa; just to add information and avoid useless computations. Anyway, the idea to use more direct membeship tests other than the GAP standard membership test is surely not optimal if many membership tests are made on the same group `G`; indeed, the "slow part" of the GAP membership test is the just computation of the GAP object and its composition tree. We can decide to adopt just the GAP membership test for everything and reduce the lines of the function `lies_in`.
3. At the moment, all the groups I'm interested in are already supported by GAP, all tho sometimes not in the way I like (e.g. the Omega groups are not subgroups of GO / SO, because they preserve different quadratic forms). The only exception is, of course, the groups with base_ring not supported by GAP.

Reply:
Concerning the relations between GO, SO, and Omega in GAP, see the discussion at gap-system/gap/issues/500.
That is, GAP should provide the possibility to create the isometry group w.r.t. a given form.

Reply:
Yeah, it is just a matter to compute a conjugate subgroup. I would like to define the standard copies of GO and SO to preserve the same form as Omega (that is the same of Magma).